DNN Platform/Modules/DDRMenu/Common/DNNContext.cs

```diff
@@ -40,73 +42,122 @@ public DNNContext(Control hostControl)
         {
             get { return (DNNContext)HttpContext.Current.Items[DataName]; }
             private set { HttpContext.Current.Items[DataName] = value; }
         }
 
+        /// <summary>
+        /// Gets a reference to the page.
+        /// </summary>
         public Page Page
         {
-            get { return this._Page ?? (this._Page = this.HostControl.Page); }
+            get { return this.page ?? (this.page = this.HostControl.Page); }
         }
 
+        /// <summary>
+        /// Gets the current portal settings.
+        /// </summary>
         public PortalSettings PortalSettings
         {
-            get { return this._PortalSettings ?? (this._PortalSettings = PortalController.Instance.GetCurrentPortalSettings()); }
+            get { return this.portalSettings ?? (this.portalSettings = (PortalSettings)PortalController.Instance.GetCurrentSettings()); }
         }
 
+        /// <summary>
+        /// Gets the currently active tab (page).
+        /// </summary>
         public TabInfo ActiveTab
         {
-            get { return this._ActiveTab ?? (this._ActiveTab = this.PortalSettings.ActiveTab); }
+            get { return this.activeTab ?? (this.activeTab = this.PortalSettings.ActiveTab); }
         }
 
+        /// <summary>
+        /// Gets the path to the skin (theme).
+        /// </summary>
         public string SkinPath
         {
-            get { return this._SkinPath ?? (this._SkinPath = this.ActiveTab.SkinPath); }
+            get { return this.skinPath ?? (this.skinPath = this.ActiveTab.SkinPath); }
         }
 
+        /// <summary>
+        /// Gets the host control.
+        /// </summary>
         public Control HostControl { get; private set; }
 
         private static string DataName
         {
-            get { return _DataName ?? (_DataName = "DDRMenu.DNNContext." + ModuleName); }
+            get { return dataName ?? (dataName = "DDRMenu.DNNContext." + ModuleName); }
         }
 
+        /// <summary>
+        /// Converts a url into one that is usable on the requesting Client.
+        /// </summary>
+        /// <param name="relativeUrl">The relative url.</param>
+        /// <returns>The converted url.</returns>
         public string ResolveUrl(string relativeUrl)
         {
             return this.HostControl.ResolveUrl(relativeUrl);
         }
 
+        /// <inheritdoc/>
         public void Dispose()
         {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Disposes this instance resources.
+        /// </summary>
+        /// <param name="disposing">A value indicating if the current instance is disposing.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (this.isDisposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                // free managed resources.
+                this.page.Dispose();

```

Comment:
I'm not familiar with this `DNNContext` class, but I doubt that it's responsible for disposing of the passed control's `Page` reference.

Reply:
Ok, it's been addressed in 6af2a81